src/test/java/igym/entities/UserTest.java

```diff
@@ -2,32 +2,77 @@
 
 import static org.junit.jupiter.api.Assertions.*;
 
+import java.util.Set;
+import igym.config.ValidationConfig;
+
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
+import jakarta.validation.*;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+
 public class UserTest {
 
+    Validator validator = ValidationConfig.validator();
+
     @Test
-    @DisplayName("Getting user name")
+    @DisplayName("Should return the name of the user")
     public void userNameTest() {
-        String name1 = "Maria Clown";
-        User user1 = new User();
-        user1.setName(name1);
-        assertEquals(user1.getName(), name1);
-
-        String name2 = "John Textor";
-        User user2 = new User(name2);
-        assertEquals(user2.getName(), name2);
+        String userName = "Maria Clown";
+        User user = new User();
+        user.setName(userName);
+        assertEquals(user.getName(), userName);
 
-        User user3 = new User();
-        assertNull(user3.getName());
+        Set<ConstraintViolation<User>> violations = validator.validate(user);
+        assertTrue(violations.isEmpty());
     }
 
     @Test
-    @DisplayName("Testing ID from User")
+    @DisplayName("Should return ID null when an User is created")
     public void userIdTest() {
         String name1 = "Maria Clown";
         User user1 = new User(name1);
         assertNull(user1.getId(), "When a User is created, the ID should be null");
     }
+
+    @Test
+    @DisplayName("Should return violation error due to null name")
+    public void nullConstraintTest() {
+        User nullNameUser = new User();
+        Set<ConstraintViolation<User>> violations = validator.validate(nullNameUser);
+        assertTrue(violations.size() == 1);
+        ConstraintViolation<User> violation = violations.iterator().next();
+        assertEquals(NotBlank.class, violation.getConstraintDescriptor().getAnnotation().annotationType());
+    }
+
+    @Test
+    @DisplayName("Should return violation error due to blank name")
```

Comment:
I think we both confused in this way, this should be an empty string and this " " blank, but let's define this and change the test into our pattern